scripts/ubuntu/1-install-dependendcies.sh

```diff
@@ -17,10 +17,42 @@
 set -e
 
 # Install Docker Package Repo
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
+echo
+echo "####################################"
+echo "Install Docker Package Repo"
+echo "####################################"
+echo
+timeout 300s bash -c '
+start_time=$(date)
+echo "Ping started at: $start_time"
+while :; do
+  if ping -c 1 docker.download.com | grep -q "1 received"; then
+    echo "Ping docker.download.com successful"
+    end_time=$(date)
+    echo "Ping ended at: $end_time"
+    echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+    break
+  fi
+  echo "Ping docker.download.com failed, retrying..."
+  sleep 1
+done
+if [ $? -eq 124 ]; then
```

Comment:
Hi @raul-marquez-csa ,

Could you explain why you are testing for this specific return value?

Reply:
124 is the timeout reached code

Reply:
Is it possible to still merge this to the spring branch? There's an upcoming rapid recert training in June that uses the sprint release and would help not having this issue as there will be many members to support.